stat/distuv/pareto_test.go

```diff
@@ -0,0 +1,117 @@
+// Copyright ©2017 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package distuv
+
+import (
+	"sort"
+	"testing"
+
+	"golang.org/x/exp/rand"
+	"gonum.org/v1/gonum/floats"
+)
+
+func TestParetoProb(t *testing.T) {
+	for _, test := range []struct {
+		x, xm, alpha, want float64
+	}{
+		{0, 1, 1, 0},
+		{1, 1, 1, 1.0},
+		{2, 1, 1, 0.25},
+		{3, 1, 1, 0.1111111111111111},
+		{4, 1, 1, 0.0625},
+		{5, 1, 1, 0.04},
+
+		{0, 1, 2, 0},
+		{1, 1, 2, 2},
+		{2, 1, 2, 0.25},
+		{3, 1, 2, 0.074074074074074},
+		{4, 1, 2, 0.03125},
+		{5, 1, 2, 0.016},
+
+		{0, 1, 3, 0},
+		{1, 1, 3, 3.0},
+		{2, 1, 3, 0.1875},
+		{3, 1, 3, 0.037037037037037},
+		{4, 1, 3, 0.011718750000000},
+		{5, 1, 3, 0.0048},
+	} {
+		pdf := Pareto{test.xm, test.alpha, nil}.Prob(test.x)
+		if !floats.EqualWithinAbsOrRel(pdf, test.want, 1e-10, 1e-10) {
+			t.Errorf("Pdf mismatch, x = %v, xm = %v, alpha = %v. Got %v, want %v", test.x, test.xm, test.alpha, pdf, test.want)
+		}
+	}
+}
+
+func TestParetoCDF(t *testing.T) {
+	for _, test := range []struct {
+		x, xm, alpha, want float64
+	}{
+		{0, 1, 1, 0},
```

Comment:
Try some non-integer values in here too please.

Reply:
done.